crates/bevy_ui/src/widget/image.rs

```diff
@@ -27,10 +27,13 @@ pub fn image_node_system(
             .and_then(|material| material.texture.as_ref())
             .and_then(|texture_handle| textures.get(texture_handle))
         {
-            calculated_size.size = Size {
+            let size = Size {
                 width: texture.size.width as f32,
                 height: texture.size.height as f32,
             };
+            if size != calculated_size.size {
```

Comment:
Its probably a good idea to leave a comment here about why we're doing this check. People with less context might look at this and say "this check isn't needed".